src/psyclone/tests/psyir/backend/visitor_test.py

```diff
@@ -138,12 +139,66 @@ def test_psyirvisitor_visit_arg_error():
 
     '''
     visitor = PSyIRVisitor(indent_string=" ", initial_indent_depth=4)
-    with pytest.raises(VisitorError) as excinfo:
+    with pytest.raises(TypeError) as excinfo:
         visitor("hello")
+    assert ("The PSyIR visitor functor method only accepts a PSyIR Node "
+            "as argument, but found 'str'." in str(excinfo.value))
+    with pytest.raises(VisitorError) as excinfo:
+        visitor._visit("hello")
     assert ("Visitor Error: Expected argument to be of type 'Node' but found "
             "'str'." in str(excinfo.value))
 
 
+def test_psyirvisitor_lower_dsl_concepts():
+    ''' Test that DSL concepts are lowered by the visitors but the node is not
+    modified after the visitor has finished. '''
+
+    class MyDSLNode(Statement):
+        ''' DSL Concept that lowers to a return statement '''
+        _text_name = "MyDSLNode"
+
+        def lower_to_language_level(self):
+            ''' MyDSLNode lowers to a return statement and adds a symbol
+            if it is inside an scoping region. '''
+            self.scope.symbol_table.add(DataSymbol("val", REAL_TYPE))
```

Comment:
If it's not inside a scoping region, will this break?

Reply:
It will, but it is expected, it is what triggers the VisitorError checked at the end of this test. If the statement was guarded by a condition the next replace_with will also break creating the same error.

I added a comment to explain that we intent to cause this error during the test.